polus-image-assembler-plugin/src/main.py

```diff
@@ -0,0 +1,296 @@
+                out_dict['name'] += '-'
+                out_dict['name'] += ''.join([str(ind) for ind in maxvals])
+                out_dict['name'] += '>'
+                if i ==  indices[-1]:
+                    out_dict['name'] += name[indices[-1]+1:]
+                else:
+                    out_dict['name'] += name[indices[v]+1:indices[v+1]]
+                minvals = []
+                maxvals = []
+    else:
+        out_dict['name'] = name
+
+    return out_dict
+
+if __name__=="__main__":
+    # Initialize the logger
+    logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                        datefmt='%d-%b-%y %H:%M:%S')
+    logger = logging.getLogger("main")
+    logger.setLevel(logging.INFO)
+    
+    # Setup the argument parsing
+    parser = argparse.ArgumentParser(prog='main', description='Assemble images from a single stitching vector.')
+    parser.add_argument('--stitchPath', dest='stitchPath', type=str,
+                        help='Complete path to a stitching vector', required=False)
+    parser.add_argument('--imgPath', dest='imgPath', type=str,
+                        help='Input image collection to be processed by this plugin', required=True)
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    parser.add_argument('--timesliceNaming', dest='timesliceNaming', type=str,
+                        help='Use timeslice number as image name', required=False)
+    parser.add_argument('--vectorInMetadata', dest='vectorInMetadata', type=str,
+                        help='Use stitching vectors stored in the metadata', required=False)
+
+    # Parse the arguments
+    args = parser.parse_args()
+    vectorInMetadata = args.vectorInMetadata == 'true'
+    logger.info('vectorInMetadata: {}'.format(vectorInMetadata))
+    imgPath = args.imgPath
+    outDir = args.outDir
+    logger.info('outDir: {}'.format(outDir))
+    timesliceNaming = args.timesliceNaming == 'true'
+    logger.info('timesliceNaming: {}'.format(timesliceNaming))
+    if vectorInMetadata:
+        stitchPath = str(Path(imgPath).parent.joinpath('metadata_files').absolute())
+    else:
+        stitchPath = args.stitchPath
+        if stitchPath == None:
+            ValueError('If vectorInMetadata==False, then stitchPath must be defined')
+
+    # Get a list of stitching vectors
+    try:
+        vectors = [str(p.absolute()) for p in Path(stitchPath).iterdir() if p.is_file() and "".join(p.suffixes)=='.txt']
+    except FileNotFoundError:
+        # Workaround for WIPP bug
+        if Path(stitchPath).name == 'metadata_files':
+            stitchPath = str(Path(imgPath).joinpath('metadata_files').absolute())
+            imgPath = str(Path(imgPath).joinpath('images').absolute())
+            vectors = [str(p.absolute()) for p in Path(stitchPath).iterdir() if p.is_file() and "".join(p.suffixes)=='.txt']
```

Comment:
If this is a reported bug in their repo maybe link to that bug so its more clear why this exists. Just reading it, it looks like there are files that you want to not "see" but it'd be good and easier to just link to the github issue if possible.

Reply:
At the time I wrote this, it was considered a bug. It is now just standard practice to always detect what input folder you are being fed and switch to an `images` subdirectory if it's present.